packages/vuetify/src/components/VIconBtn/VIconBtn.tsx

```diff
@@ -0,0 +1,192 @@
+import { VProgressCircular } from '@/components/VProgressCircular'
+
+// Composables
+import { makeBorderProps, useBorder } from '@/composables/border'
+import { makeComponentProps } from '@/composables/component'
+import { makeElevationProps, useElevation } from '@/composables/elevation'
+import { useProxiedModel } from '@/composables/proxiedModel'
+import { makeRoundedProps, useRounded } from '@/composables/rounded'
+import { makeTagProps } from '@/composables/tag'
+import { makeThemeProps, provideTheme } from '@/composables/theme'
+import { genOverlays, makeVariantProps, useVariantFast } from '@/composables/variant'
+
+// Utilities
+import { toDisplayString } from 'vue'
+import { convertToUnit, genericComponent, propsFactory, useRender } from '@/util'
+
+// Types
+import type { PropType } from 'vue'
+import type { IconValue } from '@/composables/icons'
+import type { Variant } from '@/composables/variant'
+
+export type VIconBtnSlots = {
+  default: never
+  loader: never
+}
+
+export const makeVIconBtnProps = propsFactory({
+  active: {
+    type: Boolean,
+    default: undefined,
+  },
+  activeColor: String,
+  activeIcon: [String, Function, Object] as PropType<IconValue>,
+  activeVariant: String as PropType<Variant>,
+  baseVariant: {
+    type: String as PropType<Variant>,
+    default: 'tonal',
+  },
+  disabled: Boolean,
+  height: [Number, String],
+  width: [Number, String],
+  hideOverlay: Boolean,
+  icon: [String, Function, Object] as PropType<IconValue>,
+  iconColor: String,
+  loading: Boolean,
+  opacity: [Number, String],
+  readonly: Boolean,
+  rotate: [Number, String],
+  size: [Number, String],
+  text: {
+    type: [String, Number, Boolean],
+    default: undefined,
+  },
+
+  ...makeBorderProps(),
+  ...makeComponentProps(),
+  ...makeElevationProps(),
+  ...makeRoundedProps(),
+  ...makeTagProps({ tag: 'button' }),
+  ...makeThemeProps(),
+  ...makeVariantProps({ variant: 'flat' } as const),
+}, 'VIconBtn')
+
+export const VIconBtn = genericComponent<VIconBtnSlots>()({
+  name: 'VIconBtn',
+
+  props: makeVIconBtnProps(),
+
+  emits: {
+    'update:active': (value: boolean) => true,
+  },
+
+  setup (props, { attrs, slots }) {
+    const isActive = useProxiedModel(props, 'active')
+
+    const { themeClasses } = provideTheme(props)
+    const { borderClasses } = useBorder(props)
+    const { elevationClasses } = useElevation(props)
+    const { roundedClasses } = useRounded(props)
+
+    const { colorClasses, colorStyles, variantClasses } = useVariantFast(() => ({
+      color: props.disabled ? undefined
+      // Use an inline fallback as opposed to setting a default color
+      // because non-toggle buttons are default flat whereas toggle
+      // buttons are default tonal and active flat. The exact use
+      // case for this is a toggle button with no active color.
+      : isActive.value !== undefined && isActive.value ? props.activeColor ?? props.color ?? 'surface-variant'
+      : props.color,
```

Comment:
Was this supposed to be

```suggestion
      color: props.disabled ? undefined
      : isActive.value === undefined ? props.color
      // Use an inline fallback as opposed to setting a default color
      // because non-toggle buttons are default flat whereas toggle
      // buttons are default tonal and active flat. The exact use
      // case for this is a toggle button with no active color.
      : isActive.value ? props.activeColor ?? props.color ?? 'surface-variant'
      : props.color,
```

because they're both redundant, if `isActive.value === undefined` then the second case would be false and it would end up as `props.color` anyway.